Origin: https://github.com/dell/dkms/pull/130
Author: Thorsten Glaser <t.glaser@tarent.de>, Gianfranco Costamagna <locutusofborg@debian.org>
Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=956245
From 1156c1810cb42cda1409b36a44339aa3780dbf2f Mon Sep 17 00:00:00 2001
From: Thorsten Glaser <t.glaser@tarent.de>
Date: Mon, 15 Jun 2020 11:09:59 +0200
Subject: [PATCH] dkms: don't use UID to check for root user, but use instead
 id -u that returns the effective user.

This patch is inspired from the hint of Thorsten Glaser <t.glaser@tarent.de>
in Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=956245
---
 dkms | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dkms b/dkms
index 15c08ef..8233cf3 100644
--- a/dkms
+++ b/dkms
@@ -3017,7 +3017,7 @@ debian_install()
     done
     # if they are already installed, we are OK.
     [[ ${package[@]} ]] || return
-    if ((UID != 0)); then
+    if (($(id -u) != 0)); then
         # figure out how to get root
         for getroot in su-to-root gksudo kdesu sudo; do
             which $getroot >/dev/null 2>&1 || continue
@@ -3259,7 +3259,7 @@ parse_moduleversion(){
 }
 
 check_root() {
-    ((UID == 0)) && return
+    [[ $(id -u) = 0 ]] && return
     die 1 $"You must be root to use this command."
 }
 
@@ -3914,7 +3914,7 @@ for action_to_run in $action; do
         show_status
         ;;
     ldtarball) # Make sure they're root if we're using --force
-        if ((UID != 0)) && [[ $force = true ]]; then
+        if (($(id -u) != 0)) && [[ $force = true ]]; then
             die 1 $"You must be root to use this command with the --force option."
         fi
         load_tarball && add_module
